components/apps.js

```diff
@@ -156,6 +156,132 @@ class SteamUserApps extends SteamUserAppAuth {
 		});
 	}
 
+	/**
+	 * @param {apps: Object<string, {changenumber: number, missingToken: boolean, appinfo: object}>, packages: Object<string, {changenumber: number, missingToken: boolean, packageinfo: object}>} - Response from getProductInfo
+	 * @returns {Promise}
+	 * @protected
+	 */
+	_saveProductInfo({ apps, packages }) {
+		if (!this.options.savePicsCache) {
+			return Promise.resolve([]);
+		}
+
+		let toSave = {};
+
+		for (let appid in apps) {
+			// Public only apps are weird...
+			if (apps[appid].missingToken && !apps[appid].appinfo.public_only) {
```

Comment:
I'm really conflicted about this and would love your input. Should I save product infos with missing tokens?
Initially, I would say yes, but that will lead to a difficult to handle situation: when you activate a new product on steam, it will not update your cached product info. Currently, it only updates saved pics cache when changes have been detected since last synced `changenumber`.
Additionally, there is the case of `public_only` apps. These are weird, since even if you own it, you cannot seem to request product access tokens for them?